### ISSUE TITLE
upload_to_testflight hangs during upload

### ISSUE THREAD

**Issue by bte74** [fastlane/fastlane#18286]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
After the pipeline is executed, upload starts but stops somewhere before completion with a message which I think indicates a failure. This is not picked up by update_to_testflight and pipeline holds until being killed by CICD timeout.

##### Command executed
This is the .gitlab-ci.yml file:

build_ios:
  stage: build
  rules:
    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
  script:
    - npx pod-install
    - bundle exec fastlane ios beta password:$CI_TESTFLIGHT_PASSWORD --verbose
  artifacts:
    paths:
      - app.ipa
  tags:
    - ios

This is the corresponding fastfile:

lane :beta do |options|
    begin
      increment_build_number(xcodeproj: "ios/app.xcodeproj")
      build_app(workspace: "ios/app.xcworkspace", scheme: "app")
      ENV['FASTLANE_USER'] = 'cicd@company.com'
      ENV['FASTLANE_PASSWORD'] = options[:password]
      upload_to_testflight(
        username: ENV['FASTLANE_USER'],
        app_identifier: 'com.app.id',
        apple_id: '0123456789',
        skip_waiting_for_build_processing: false
      )
    end

##### Complete output when running fastlane, including the stack trace and command used
The following is the output after build is complete and signed:

INFO [2021-02-25 20:32:59.37]: Successfully exported and compressed dSYM file
INFO [2021-02-25 20:32:59.39]: Successfully exported and signed the ipa file:
INFO [2021-02-25 20:32:59.39]: /some/path/to/app.ipa
INFO [2021-02-25 20:32:59.60]: ----------------------------------
INFO [2021-02-25 20:32:59.60]: --- Step: upload_to_testflight ---
INFO [2021-02-25 20:32:59.60]: ----------------------------------
INFO [2021-02-25 20:32:59.61]: Ready to upload new build to TestFlight (App: 0123456789)...
DEBUG [2021-02-25 20:32:59.61]: App Platform (ios)
INFO [2021-02-25 20:33:06.45]: Wrote XML data to '/var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp'
INFO [2021-02-25 20:33:06.49]: Going to upload updated app to App Store Connect
INFO [2021-02-25 20:33:06.49]: This might take a few minutes. Please don't interrupt the script.
DEBUG [2021-02-25 20:33:06.49]: ITMS_TRANSPORTER_PASSWORD=YourPassword xcrun iTMSTransporter -m upload -u cicd@company.com -p @env:ITMS_TRANSPORTER_PASSWORD -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000 2>&1
DEBUG [2021-02-25 20:33:16.21]: [Transporter]: INFO: Configuring logging...
DEBUG [2021-02-25 20:33:16.22]: [Transporter]: INFO: Logging level set to eXtreme
DEBUG [2021-02-25 20:33:16.23]: [Transporter]: INFO: Transporter is searching for new software components.
DEBUG [2021-02-25 20:33:16.23]: [Transporter]: INFO: INFO: using cached repository.xml file.
DEBUG [2021-02-25 20:33:16.58]: [Transporter]: INFO: Update check complete.
DEBUG [2021-02-25 20:33:20.71]: [Transporter]: DEBUG: Attempting refresh of configuration data from https://contentdelivery.itunes.apple.com/transporter/Defaults.properties
DEBUG [2021-02-25 20:33:21.49]: [Transporter]: DEBUG: Configuration refresh successful.
DEBUG [2021-02-25 20:33:21.49]: [Transporter]: DEBUG: Saving configuration to local path: /Users/MacUserAccountName/Library/Caches/com.apple.amp.itmstransporter/Defaults.properties
DEBUG [2021-02-25 20:33:21.80]: [Transporter]: INFO: iTMSTransporter : iTunes Store Transporter [2.1.0]
DEBUG [2021-02-25 20:33:21.80]: [Transporter]: INFO: OS identifier: Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:21.80]: [Transporter]: INFO: Memory: [JVM] 995M free, 1075M total, 1820M max [System] (Physical) 1522M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:21.81]: [Transporter]: DBG-X: Memory: [JVM] 995M free, 1075M total, 1820M max [System] (Physical) 1518M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:21.81]: [Transporter]: DBG-X: Using operation named: authenticateForSession
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter StatisticsClientStartDateTimeZoneISO = 2021-02-25T20:33:21+03:00
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter StatisticsPreviousCallDurationInSecs = 1.11134675
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:21.83]: [Transporter]: INFO: id = 20210225203321-117
DEBUG [2021-02-25 20:33:21.84]: [Transporter]: INFO: iTMSTransporter Correlation Key: c526c20d-25df-4dd3-82dc-4b635420e471-0001
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter SharedSecret = **hidden value**
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter MultipartUploadsEnabled = true
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter TxHeaders = {jenga=true}
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter SessionExpiration = 2021-03-01T17:33:22.735Z
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter SessionId = CPaqDBIQ0YZXYtGCRI+4mmomGqpuLA==
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter ShouldUseRESTAPIs = false
DEBUG [2021-02-25 20:33:22.89]: [Transporter]: DBG-X:   parameter StreamingSettings = {LogStreamingEnabled=true, MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false}
DEBUG [2021-02-25 20:33:22.94]: [Transporter]: INFO: Upload mode selected.
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X: Memory: [JVM] 1041M free, 1090M total, 1820M max [System] (Physical) 1489M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X: Using operation named: authenticateForTransportDiscovery
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter StatisticsClientStartDateTimeZoneISO = 2021-02-25T20:33:22+03:00
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: INFO: id = 20210225203322-371
DEBUG [2021-02-25 20:33:22.95]: [Transporter]: INFO: iTMSTransporter Correlation Key: c526c20d-25df-4dd3-82dc-4b635420e471-0001
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter PreferredTransports = [HTTP, Signiant, Aspera, DAV]
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter TransportHosts.Signiant = [sgr602.apple.com:44001, sgr603.apple.com:44001, sgr601.apple.com:44001]
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter SessionExpiration = 2021-03-01T17:33:23.725Z
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter TransportHosts.DAV = [itmsdav.apple.com:443]
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter TransportHosts.Aspera = [vgr501.apple.com:33001, vgr503.apple.com:33001, vgr504.apple.com:33001, vgr502.apple.com:33001]
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter ShouldUseRESTAPIs = false
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DBG-X:   parameter StreamingSettings = {LogStreamingEnabled=true, MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false}
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Transporter's command line arguments are: -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Examining the package at: /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Ensuring that package has well formed metadata file...
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DEBUG: Getting case-sensitive metadata xml filename...
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Gathering the list of valid files from the package ...
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DEBUG:   metadata.xml will be verified by Apple's web service to determine if it is a valid file.
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DEBUG:   0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa will be verified by Apple's web service to determine if it is a valid file.
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Finished gathering the list of valid files from the package.
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: DEBUG:   Case-sensitive metadata filename is: metadata.xml
DEBUG [2021-02-25 20:33:23.93]: [Transporter]: INFO: Performing authentication of package 0123456789.itmsp ...
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X: Memory: [JVM] 1029M free, 1090M total, 1820M max [System] (Physical) 1475M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X: Deflated 485 bytes to 424
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X: Using operation named: validateMetadata
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter Files = [metadata.xml, 0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa]
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter Metadata = <?xml version="1.0" encoding="UTF-8"?>
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter MetadataChecksum = f1ab2427f42e05a0fa25a11450f3fa07
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter MetadataCompressed = (suppressed)
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter MetadataInfo = {app_platform=ios, primary_bundle_identifier=, device_id=, bundle_identifier=, packageVersion=software5.4, apple_id=0123456789, asset_types=[bundle], bundle_version=, bundle_short_version_string=}
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter PackageName = 0123456789.itmsp
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter PackageSize = 159829477
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter StatisticsClientStartDateTimeZoneISO = 2021-02-25T20:33:24+03:00
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter Throttle = 100000
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: INFO: id = 20210225203324-307
DEBUG [2021-02-25 20:33:24.01]: [Transporter]: INFO: iTMSTransporter Correlation Key: c526c20d-25df-4dd3-82dc-4b635420e471-0001
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter SessionExpiration = 2021-03-01T17:33:24.162Z
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter MinimumFileSizeThresholdForCheckum = 104857600
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter SoftwareMediaDescriptionFormat = binary
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter SoftwareMediaDescriptionThresholds = {in-app-purchase-content={extensions=[pkg], size=100000000}, ccats={extensions=[pdf], size=100000000}, product-archive={extensions=[pkg], size=100000000}, developer-id-package={extensions=[zip, pkg, dmg], size=100000000}, bundle={extensions=[zip, ipa], size=100000000}, large-icon={extensions=[jpg, png, jpeg], size=100000000}}
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter Success = true
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter StreamingSettings = {LogStreamingEnabled=true, MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false}
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter FilesToUpload = [metadata.xml, 0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa]
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter AssetsToDescribe = [{Role=source, Filename=0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa, ShouldDescribeAsset=true, UTI=com.apple.ipa}]
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter CPUToken = 9ffed06e-150a-449f-865b-6f4c42fe6c74/1614274404586
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter Checksums = {0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa=bdf3c388d3a9190a66f7fec31f7e6f75}
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter FileSizes = {0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa=159828992}
DEBUG [2021-02-25 20:33:24.69]: [Transporter]: DBG-X:   parameter ShouldUseRESTAPIs = false
DEBUG [2021-02-25 20:33:24.70]: [Transporter]: DBG-X:   parameter ProviderNameForToken = 45NBAM4WJ6
DEBUG [2021-02-25 20:33:24.71]: [Transporter]: INFO: The list of files requested for upload by Apple is: [metadata.xml, 0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa].  These are the files that will be uploaded.
DEBUG [2021-02-25 20:33:24.71]: [Transporter]: INFO: Starting media analysis of assets
DEBUG [2021-02-25 20:33:24.93]: [Transporter]: DEBUG: Handling asset: filename=0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa,uti=com.apple.ipa,role=source,describe=true
DEBUG [2021-02-25 20:33:24.93]: [Transporter]: DEBUG: Using container type tool name and version: iOS App Describer:1.26
DEBUG [2021-02-25 20:33:24.93]: [Transporter]: DEBUG: Using path: /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp/0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa
DEBUG [2021-02-25 20:33:43.35]: [Transporter]: DBG-X: description length:0
DEBUG [2021-02-25 20:33:43.35]: [Transporter]: DBG-X: Asset description: 
DEBUG [2021-02-25 20:33:43.35]: [Transporter]: DEBUG: Asset successfully described: com.apple.jingle.leghorn.software.AbstractSoftwareDescriber$1@7a04fea7
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X: Deflated 646 bytes to 411
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: INFO: Asset media analysis has completed
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X: Memory: [JVM] 1043M free, 1091M total, 1820M max [System] (Physical) 1388M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X: Using operation named: lookupTransportDiagnostic
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter StatisticsClientStartDateTimeZoneISO = 2021-02-25T20:33:43+03:00
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: INFO: id = 20210225203343-686
DEBUG [2021-02-25 20:33:43.56]: [Transporter]: INFO: iTMSTransporter Correlation Key: c526c20d-25df-4dd3-82dc-4b635420e471-0001
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter AsperaLoggingLevel = OFF
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter SigniantLoggingLevel = OFF
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter SessionExpiration = 2021-03-01T17:33:43.718Z
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter ShouldUseRESTAPIs = false
DEBUG [2021-02-25 20:33:43.95]: [Transporter]: DBG-X:   parameter StreamingSettings = {LogStreamingEnabled=true, MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false}
DEBUG [2021-02-25 20:33:43.98]: [Transporter]: DBG-X: Using operation named: createTestReservation
DEBUG [2021-02-25 20:33:43.98]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter fileDescriptions = ( one FileDescription )
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter NewPackageName = http-transport-diagnostic-20210225203343-EET-8033.itmsp
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter Username = cicd@company.com
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:43.99]: [Transporter]: INFO: id = 20210225203343-969
DEBUG [2021-02-25 20:33:44.10]: [Transporter]: INFO: iTMSTransporter Correlation Key: 0006f42d-3888-40fb-a838-7c5e040086f3-0001
DEBUG [2021-02-25 20:33:45.20]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:45.20]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:45.20]: [Transporter]: DBG-X:   parameter Reservations = ( one Reservation )
DEBUG [2021-02-25 20:33:45.20]: [Transporter]: DBG-X:   parameter StreamingSettings = {MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false, LogStreamingEnabled=true}
DEBUG [2021-02-25 20:33:45.20]: [Transporter]: DBG-X:   parameter Success = true
DEBUG [2021-02-25 20:33:45.21]: [Transporter]: INFO: Setting transport log file: /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/http-transport-diagnostic-20210225203343-EET-8033.itmsp5221766342448174617.log
DEBUG [2021-02-25 20:33:45.21]: [Transporter]: INFO: Starting upload for packageName: http-transport-diagnostic-20210225203343-EET-8033.itmsp
DEBUG [2021-02-25 20:33:45.21]: [Transporter]: INFO: Thread Pool Configuration: {rateLimit=100000 KBs, minConnections=6, coreThreads=6, perCoreThreadMultiplier=1.65, processors=4, maxThreads=7, maxConnections=6}
DEBUG [2021-02-25 20:33:45.22]: [Transporter]: INFO: Reservation CPUSource114/v4/9d/32/a8/9d32a88b-f545-60d2-753d-56b695d86942/http-transport-diagnostic-6851420517690560029.txt created for file: (http-transport-diagnostic-6851420517690560029.txt), part count:(1), file size: (199)
DEBUG [2021-02-25 20:33:45.22]: [Transporter]: INFO: Computing total size of files to upload ...
DEBUG [2021-02-25 20:33:45.22]: [Transporter]: DBG-X:   adding file for % complete logging: http-transport-diagnostic-6851420517690560029.txt , size: 199
DEBUG [2021-02-25 20:33:45.22]: [Transporter]: DEBUG: Setting total size of files for upload to: 199
DEBUG [2021-02-25 20:33:45.22]: [Transporter]: INFO: Done computing total size of files to upload.
DEBUG [2021-02-25 20:33:45.24]: [Transporter]: INFO: Starting upload for reservation: (CPUSource114/v4/9d/32/a8/9d32a88b-f545-60d2-753d-56b695d86942/http-transport-diagnostic-6851420517690560029.txt) file: (http-transport-diagnostic-6851420517690560029.txt/1) size: 199
DEBUG [2021-02-25 20:33:46.21]: [Transporter]: INFO:      File: http-transport-diagnostic-6851420517690560029.txt 199/199, 100.00% completed
DEBUG [2021-02-25 20:33:46.21]: [Transporter]: INFO: Package upload progress: 100.00% completed
DEBUG [2021-02-25 20:33:46.46]: [Transporter]: INFO: Finished part upload for: (http-transport-diagnostic-6851420517690560029.txt/1) 777.34 B/s in 0 seconds
DEBUG [2021-02-25 20:33:46.46]: [Transporter]: INFO: Finished upload for reservation: (CPUSource114/v4/9d/32/a8/9d32a88b-f545-60d2-753d-56b695d86942/http-transport-diagnostic-6851420517690560029.txt) file: (http-transport-diagnostic-6851420517690560029.txt)
DEBUG [2021-02-25 20:33:46.46]: [Transporter]: INFO: Waiting on reservation commit(s) to finish.
DEBUG [2021-02-25 20:33:46.46]: [Transporter]: DBG-X: Using operation named: commitTestReservation
DEBUG [2021-02-25 20:33:46.46]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter NewPackageName = http-transport-diagnostic-20210225203343-EET-8033.itmsp
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter reservations = [CPUSource114/v4/9d/32/a8/9d32a88b-f545-60d2-753d-56b695d86942/http-transport-diagnostic-6851420517690560029.txt]
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter Username = cicd@company.com
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: INFO: id = 20210225203346-180
DEBUG [2021-02-25 20:33:46.47]: [Transporter]: INFO: iTMSTransporter Correlation Key: 0006f42d-3888-40fb-a838-7c5e040086f3-0001
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: DBG-X:   parameter StreamingSettings = {MultipartUploadsEnabled=true, AssetDescriptionStreamingEnabled=false, LogStreamingEnabled=true}
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: DBG-X:   parameter Success = true
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: INFO: Reservation committed: (CPUSource114/v4/9d/32/a8/9d32a88b-f545-60d2-753d-56b695d86942/http-transport-diagnostic-6851420517690560029.txt) file: (http-transport-diagnostic-6851420517690560029.txt)
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: INFO: All reservation commit(s) finished.
DEBUG [2021-02-25 20:33:46.82]: [Transporter]: INFO: 124.45 B/s in 1 second
DEBUG [2021-02-25 20:33:46.83]: [Transporter]: DEBUG: Using token file path /Users/MacUserAccountName/Library/Caches/com.apple.amp.itmstransporter/UploadTokens/iMAC_45NBAM4WJ6_0123456789.itmsp.token
DEBUG [2021-02-25 20:33:46.83]: [Transporter]: DEBUG: Attempting exclusive file lock on token file
DEBUG [2021-02-25 20:33:46.83]: [Transporter]: DEBUG: Token exists, examining
DEBUG [2021-02-25 20:33:46.90]: [Transporter]: DEBUG: Token pid: 86871, found process = false; my pid: 1733
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DEBUG: Reusing existing token: 7834498e-dbf9-4c36-9f08-85db2447a61d/1614247961579
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DEBUG: Set token: 7834498e-dbf9-4c36-9f08-85db2447a61d/1614247961579
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X: Memory: [JVM] 1014M free, 1091M total, 1820M max [System] (Physical) 1383M free, 8192M total (Swap) 1026M free, 6656M total
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X: Deflated 485 bytes to 424
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X: Using operation named: validateAssets
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X: Apple's web service operation input parameters:
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X:   parameter Application = iTMSTransporter
DEBUG [2021-02-25 20:33:46.95]: [Transporter]: DBG-X:   parameter BaseVersion = 1.13.0
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter CPUToken = 7834498e-dbf9-4c36-9f08-85db2447a61d/1614247961579
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter Files = [metadata.xml, 0605177bd52a0d3c19b28963f5011b71ce73e08d5a4220b13392807e99bac78a.ipa]
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter MetadataChecksum = f1ab2427f42e05a0fa25a11450f3fa07
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter MetadataCompressed = (suppressed)
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter MetadataInfo = {app_platform=ios, primary_bundle_identifier=, device_id=, bundle_identifier=, packageVersion=software5.4, apple_id=0123456789, asset_types=[bundle], bundle_version=, bundle_short_version_string=}
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter OSIdentifier = Mac OS X 10.15.2 (x86_64); jvm=25.71-b02; jre=1.8.0-u131-b11-iTMSTransporter-b02
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter PackageName = 0123456789.itmsp
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter PackageSize = 159829477
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter RequestUUID = afc6b145-83dd-4b39-9925-90e86448ea50
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter StatisticsClientStartDateTimeZoneISO = 2021-02-25T20:33:46+03:00
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter StreamingInfoList = [file: 70750edf-8689-49cf-888a-cc19e9761bc32043676709189200747.swinfo.plist.gz, content-length: 1085547, content-type: application/binary, MD5: b320277487f57e021f0de869ffa560cf]
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter Throttle = 100000
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter Transport = HTTP
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter TransporterArguments = -m upload -u cicd@company.com -p **hidden value** -f /var/folders/yq/1bzptg_s6hng_p2y_h3610f80000gn/T/d20210225-91747-130yixh/0123456789.itmsp -k 100000
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter Version = 2.1.0
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: DBG-X:   parameter iTMSTransporterMode = upload
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: INFO: id = 20210225203346-735
DEBUG [2021-02-25 20:33:46.96]: [Transporter]: INFO: iTMSTransporter Correlation Key: c526c20d-25df-4dd3-82dc-4b635420e471-0001
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X: Apple's web service operation return value:
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter EnableJWTForAllCalls = false
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter MinimumFileSizeThresholdForCheckum = 104857600
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter SoftwareMediaDescriptionFormat = binary
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter RestartClient = false
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter SoftwareMediaDescriptionThresholds = {in-app-purchase-content={extensions=[pkg], size=100000000}, ccats={extensions=[pdf], size=100000000}, product-archive={extensions=[pkg], size=100000000}, developer-id-package={extensions=[zip, pkg, dmg], size=100000000}, bundle={extensions=[zip, ipa], size=100000000}, large-icon={extensions=[jpg, png, jpeg], size=100000000}}
DEBUG [2021-02-25 20:34:06.58]: [Transporter]: DBG-X:   parameter Success = false
WARN [2021-02-25 23:22:16.32]: Lane Context:
INFO [2021-02-25 23:22:16.33]: {:PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :BUILD_NUMBER=>"12", :IPA_OUTPUT_PATH=>"/Users/MacUserAccountName/builds/8X5AGdCC/0/company/mypath/myapp/myapp.ipa", :XCODEBUILD_ARCHIVE=>"/Users/MacUserAccountName/Library/Developer/Xcode/Archives/2021-02-25/myapp 2021-02-25 20.24.02.xcarchive", :DSYM_OUTPUT_PATH=>"/Users/MacUserAccountName/builds/8X5AGdCC/0/company/mypath/myapp/myapp.app.dSYM.zip"}
INFO [2021-02-25 23:22:16.33]: Successfully generated documentation at path '/Users/MacUserAccountName/builds/8X5AGdCC/0/company/mypath/myapp/fastlane/README.md'
+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | increment_build_number | 1           |
| 2    | build_app              | 551         |
| 3    | upload_to_testflight   | 10156       |
+------+------------------------+-------------+
DEBUG [2021-02-25 23:22:16.35]: All plugins are up to date
ERROR [2021-02-25 23:22:16.35]: fastlane finished with errors
DEBUG [2021-02-25 23:22:16.35]: All plugins are up to date
#######################################################################
# fastlane 2.176.0 is available. You are on 2.168.0.
# You should use the latest version.
# Please update using `bundle update fastlane`.
#######################################################################
2.176.0 Improvements
* [pilot] fix default value of `demo_account_required` to nil instead of false (#18158) via Albert Casademont
* [pilot] also wait for build processing when only distributing (#18073) via Benedek Kozma
* [scan] add possibility for copying pre-built .xctestrun file into test_output directory (#18057) via Nemanja Filipovic
* [precheck] update error message for `Precheck cannot check In-app purchases with the App Store Connect API Key` (#18251) via Lachlan
* [action] jazzy - add module_version option (#18260) via Marcin Stepnowski
* [spaceship][spaceauth] fix SPACESHIP_2FA_SMS_DEFAULT_PHONE_NUMBER forcing 'sms' push mode (#18239) via Hiroto Nakamura
* [screengrab] fix regression #18240: tries to use `adb.exeadb` as the executable name on Windows. (#18241) via Kartik Soneji
2.175.0 provements
* [action] fix swiftlint validation block. (#18030) via Roger Oba
* [screengrab][actions] - fix android executable checking on windows (#18172) via Kartik Soneji
* [match] add documentation on how to use GitHub tokens (#18066) via Andres Riofrio
* [fastlane] remove "sudo" for gem install (#18189) via Satoshi Namai
* [match] improve copy when asking user for password in interactive mode. (#18199) via Roger Oba
* [deliver] sort screenshots naturally, in a human-friendly way. (#18200) via Roger Oba
* [spaceship] allow extra fields in API Key JSON file (#18235) via Alex
* [action] download_dsyms - fix latest build lookup by ignoring the version. (#18201) via Roger Oba
* [match] nuke use also token when cloning private repo (#18202) via Alisher Ulugbekov
* [spaceship] fix: use new analytics endpoint (#18216) via Valerio Castelli
* [app store connect] unified ENV vars for App Store Connect API Key (#18181) via Josh Holtz
* [fastlane_core] env_names - new config item option (#18173) via Josh Holtz
* [action] fix download dsyms with zero prefixed versions (#18042) via Daisuke Shimamoto
## Notes
- Added new unified environment variables for App Store Connect API Keys
  - `APP_STORE_CONNECT_API_KEY` and `APP_STORE_CONNECT_API_KEY_PATH` can be used in every action/tool that uses App Store Connect API Keys
  - Action specific environment variables will continue to keep working
2.174.0 Improvements
* [action] app_store_connect_api_key update App Store Connect API token generation documentation (#18068) via David Brunow
* [action] appetize add timeout support (#18074) via Raul Riera
* [spaceship] expose error also from source (#18150) via Alisher Ulugbekov
* [spaceship] added ASC endpoints for bundle id & bundle id capability (#18080) via Dinsen
* [action] - cocoapods - add allow_root option (#18076) via Atsuto Yamashita
* [fastlane_core] improve config item validation conversion and use UI.error instead of puts to report validation exception (#18088) via Gio Lodi
* [action]  push_to_git_remote - fix "undefined method 'each' for NilClass" error (#18153) via Neil Cowburn
* [spaceship] explicitly allow to continue 2FA in non-interactive mode (#18109) via Oleg Kohtenko
* [action] carthage - add use_xcframeworks option (#18090) via Atsuto Yamashita
* [carthage] Add archive option (#18077) via Atsuto Yamashita
* [spaceship] fixed typo (#18103) via Atsuto Yamashita
* [action] xcov - fix compatibility issues for plugins depending on FastlaneCore::Project (#18053) via Satoshi Namai
To see all new releases, open https://github.com/fastlane/fastlane/releases
Please update using `bundle update fastlane`
ERROR [2021-02-25 23:22:17.62]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
ERROR: Job failed: execution took longer than 3h0m0s seconds

<details>
In short, there are two issues that I need help with:

1- I would really appreciate help in interpreting the last part of the log, the part that comes from Apple iTransporter. This is the part that ends with

parameter Success = false

I understand upload_to_testflight may not be involved here, but I would really appreciate if another set of eyes take a look at it and help me decipher what iTransporter is trying to tell me.

2- I would expect that whenever iTransporter returns an error value and stops, upload_to_testflight picks this up and returns with an error code before timeout kills the job.
</details>

### Environment

gemfile.lock from the repo:

GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (3.0.2)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    atomos (0.1.3)
    aws-eventstream (1.1.0)
    aws-partitions (1.399.0)
    aws-sdk-core (3.109.3)
      aws-eventstream (~> 1, >= 1.0.2)
      aws-partitions (~> 1, >= 1.239.0)
      aws-sigv4 (~> 1.1)
      jmespath (~> 1.0)
    aws-sdk-kms (1.39.0)
      aws-sdk-core (~> 3, >= 3.109.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-s3 (1.85.0)
      aws-sdk-core (~> 3, >= 3.109.0)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.1)
    aws-sigv4 (1.2.2)
      aws-eventstream (~> 1, >= 1.0.2)
    babosa (1.0.4)
    claide (1.0.3)
    colored (1.2)
    colored2 (3.1.2)
    commander-fastlane (4.4.6)
      highline (~> 1.7.2)
    declarative (0.0.20)
    declarative-option (0.1.0)
    digest-crc (0.6.1)
      rake (~> 13.0)
    domain_name (0.5.20190701)
      unf (>= 0.0.5, < 1.0.0)
    dotenv (2.7.6)
    emoji_regex (3.2.1)
    excon (0.78.0)
    faraday (1.1.0)
      multipart-post (>= 1.2, < 3)
      ruby2_keywords
    faraday-cookie_jar (0.0.7)
      faraday (>= 0.8.0)
      http-cookie (~> 1.0.0)
    faraday_middleware (1.0.0)
      faraday (~> 1.0)
    fastimage (2.2.0)
    fastlane (2.168.0)
      CFPropertyList (>= 2.3, < 4.0.0)
      addressable (>= 2.3, < 3.0.0)
      aws-sdk-s3 (~> 1.0)
      babosa (>= 1.0.3, < 2.0.0)
      bundler (>= 1.12.0, < 3.0.0)
      colored
      commander-fastlane (>= 4.4.6, < 5.0.0)
      dotenv (>= 2.1.1, < 3.0.0)
      emoji_regex (>= 0.1, < 4.0)
      excon (>= 0.71.0, < 1.0.0)
      faraday (~> 1.0)
      faraday-cookie_jar (~> 0.0.6)
      faraday_middleware (~> 1.0)
      fastimage (>= 2.1.0, < 3.0.0)
      gh_inspector (>= 1.1.2, < 2.0.0)
      google-api-client (>= 0.37.0, < 0.39.0)
      google-cloud-storage (>= 1.15.0, < 2.0.0)
      highline (>= 1.7.2, < 2.0.0)
      json (< 3.0.0)
      jwt (>= 2.1.0, < 3)
      mini_magick (>= 4.9.4, < 5.0.0)
      multipart-post (~> 2.0.0)
      plist (>= 3.1.0, < 4.0.0)
      rubyzip (>= 2.0.0, < 3.0.0)
      security (= 0.1.3)
      simctl (~> 1.6.3)
      slack-notifier (>= 2.0.0, < 3.0.0)
      terminal-notifier (>= 2.0.0, < 3.0.0)
      terminal-table (>= 1.4.5, < 2.0.0)
      tty-screen (>= 0.6.3, < 1.0.0)
      tty-spinner (>= 0.8.0, < 1.0.0)
      word_wrap (~> 1.0.0)
      xcodeproj (>= 1.13.0, < 2.0.0)
      xcpretty (~> 0.3.0)
      xcpretty-travis-formatter (>= 0.0.3)
    fastlane-plugin-increment_version_code (0.4.3)
    gh_inspector (1.1.3)
    google-api-client (0.38.0)
      addressable (~> 2.5, >= 2.5.1)
      googleauth (~> 0.9)
      httpclient (>= 2.8.1, < 3.0)
      mini_mime (~> 1.0)
      representable (~> 3.0)
      retriable (>= 2.0, < 4.0)
      signet (~> 0.12)
    google-cloud-core (1.5.0)
      google-cloud-env (~> 1.0)
      google-cloud-errors (~> 1.0)
    google-cloud-env (1.4.0)
      faraday (>= 0.17.3, < 2.0)
    google-cloud-errors (1.0.1)
    google-cloud-storage (1.29.1)
      addressable (~> 2.5)
      digest-crc (~> 0.4)
      google-api-client (~> 0.33)
      google-cloud-core (~> 1.2)
      googleauth (~> 0.9)
      mini_mime (~> 1.0)
    googleauth (0.14.0)
      faraday (>= 0.17.3, < 2.0)
      jwt (>= 1.4, < 3.0)
      memoist (~> 0.16)
      multi_json (~> 1.11)
      os (>= 0.9, < 2.0)
      signet (~> 0.14)
    highline (1.7.10)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    httpclient (2.8.3)
    jmespath (1.4.0)
    json (2.3.1)
    jwt (2.2.2)
    memoist (0.16.2)
    mini_magick (4.11.0)
    mini_mime (1.0.2)
    multi_json (1.15.0)
    multipart-post (2.0.0)
    nanaimo (0.3.0)
    naturally (2.2.0)
    os (1.1.1)
    plist (3.5.0)
    public_suffix (4.0.6)
    rake (13.0.1)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    retriable (3.1.2)
    rouge (2.0.7)
    ruby2_keywords (0.0.2)
    rubyzip (2.3.0)
    security (0.1.3)
    signet (0.14.0)
      addressable (~> 2.3)
      faraday (>= 0.17.3, < 2.0)
      jwt (>= 1.5, < 3.0)
      multi_json (~> 1.10)
    simctl (1.6.8)
      CFPropertyList
      naturally
    slack-notifier (2.3.2)
    terminal-notifier (2.0.0)
    terminal-table (1.8.0)
      unicode-display_width (~> 1.1, >= 1.1.1)
    tty-cursor (0.7.1)
    tty-screen (0.8.1)
    tty-spinner (0.9.3)
      tty-cursor (~> 0.7)
    uber (0.1.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.7)
    unicode-display_width (1.7.0)
    word_wrap (1.0.0)
    xcodeproj (1.19.0)
      CFPropertyList (>= 2.3.3, < 4.0)
      atomos (~> 0.1.3)
      claide (>= 1.0.2, < 2.0)
      colored2 (~> 3.1)
      nanaimo (~> 0.3.0)
    xcpretty (0.3.0)
      rouge (~> 2.0.7)
    xcpretty-travis-formatter (1.0.0)
      xcpretty (~> 0.2, >= 0.0.7)

PLATFORMS
  ruby

DEPENDENCIES
  fastlane
  fastlane-plugin-increment_version_code

BUNDLED WITH
   2.1.4

<details>
The process is initiated via gitlab CICD. The runner is running on a Mac that is local to our company but I do not have admin privileges to (I am not well versed in Mac environments anyway). The Apple account involved in the process has been opened in accordance with the instructions here:

https://docs.fastlane.tools/best-practices/continuous-integration/#separate-apple-id-for-ci

Copying the relevant part from that link:

The easiest way to get fastlane running on a CI system is to create a separate Apple ID that

doesn't have 2-factor authentication enabled
doesn't have the Account Holder role

Creating a separate Apple ID allows you to limit the permission scope (limited access to only the apps and resources it needs), have a long, randomly generated password, and will make it much more convenient for you to set up CI using fastlane.

</details>

**Comment by bte74** [comment]
<details><summary>🚫 fastlane environment 🚫</summary>
### Stack
| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.2                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.15.0                                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.2 (19C57)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.7                                                           |
### System Locale
| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |
### fastlane files:
<details><summary>`./fastlane/Fastfile`</summary>
```ruby
platform :ios do
  desc "Submit beta build to TestFlight"
lane :beta do |options|
    begin
      increment_build_number(xcodeproj: "ios/bigBen.xcodeproj")
      build_app(workspace: "ios/bigBen.xcworkspace", scheme: "bigBen")
      ENV['FASTLANE_USER'] = 'cicd@company.com'
      ENV['FASTLANE_PASSWORD'] = options[:password]
      upload_to_testflight(
        username: ENV['FASTLANE_USER'],
        app_identifier: 'com.app.id',
        apple_id: 'somenumberhere',
        skip_waiting_for_build_processing: false
      )
    end
end
```
</details>
**No Appfile found**
### fastlane gems
| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.168.0 | 🚫 Update available |
### Loaded fastlane plugins:
| Plugin                                 | Version | Update-Status |
| -------------------------------------- | ------- | ------------- |
| fastlane-plugin-increment_version_code | 0.4.3   | ✅ Up-To-Date  |
<details><summary><b>Loaded gems</b></summary>
| Gem                                    | Version      |
| -------------------------------------- | ------------ |
| did_you_mean                           | 1.3.0        |
| bundler                                | 2.1.4        |
| rake                                   | 13.0.1       |
| CFPropertyList                         | 3.0.2        |
| public_suffix                          | 4.0.6        |
| addressable                            | 2.7.0        |
| atomos                                 | 0.1.3        |
| aws-eventstream                        | 1.1.0        |
| aws-partitions                         | 1.399.0      |
| aws-sigv4                              | 1.2.2        |
| jmespath                               | 1.4.0        |
| aws-sdk-core                           | 3.109.3      |
| aws-sdk-kms                            | 1.39.0       |
| aws-sdk-s3                             | 1.85.0       |
| babosa                                 | 1.0.4        |
| claide                                 | 1.0.3        |
| colored                                | 1.2          |
| colored2                               | 3.1.2        |
| highline                               | 1.7.10       |
| commander-fastlane                     | 4.4.6        |
| declarative                            | 0.0.20       |
| declarative-option                     | 0.1.0        |
| digest-crc                             | 0.6.1        |
| unf_ext                                | 0.0.7.7      |
| unf                                    | 0.1.4        |
| domain_name                            | 0.5.20190701 |
| dotenv                                 | 2.7.6        |
| emoji_regex                            | 3.2.1        |
| excon                                  | 0.78.0       |
| multipart-post                         | 2.0.0        |
| ruby2_keywords                         | 0.0.2        |
| faraday                                | 1.1.0        |
| http-cookie                            | 1.0.3        |
| faraday-cookie_jar                     | 0.0.7        |
| faraday_middleware                     | 1.0.0        |
| fastimage                              | 2.2.0        |
| gh_inspector                           | 1.1.3        |
| jwt                                    | 2.2.2        |
| memoist                                | 0.16.2       |
| multi_json                             | 1.15.0       |
| os                                     | 1.1.1        |
| signet                                 | 0.14.0       |
| googleauth                             | 0.14.0       |
| httpclient                             | 2.8.3        |
| mini_mime                              | 1.0.2        |
| uber                                   | 0.1.0        |
| representable                          | 3.0.4        |
| retriable                              | 3.1.2        |
| google-api-client                      | 0.38.0       |
| google-cloud-env                       | 1.4.0        |
| google-cloud-errors                    | 1.0.1        |
| google-cloud-core                      | 1.5.0        |
| google-cloud-storage                   | 1.29.1       |
| json                                   | 2.3.1        |
| mini_magick                            | 4.11.0       |
| plist                                  | 3.5.0        |
| rubyzip                                | 2.3.0        |
| security                               | 0.1.3        |
| naturally                              | 2.2.0        |
| simctl                                 | 1.6.8        |
| slack-notifier                         | 2.3.2        |
| terminal-notifier                      | 2.0.0        |
| unicode-display_width                  | 1.7.0        |
| terminal-table                         | 1.8.0        |
| tty-screen                             | 0.8.1        |
| tty-cursor                             | 0.7.1        |
| tty-spinner                            | 0.9.3        |
| word_wrap                              | 1.0.0        |
| nanaimo                                | 0.3.0        |
| xcodeproj                              | 1.19.0       |
| rouge                                  | 2.0.7        |
| xcpretty                               | 0.3.0        |
| xcpretty-travis-formatter              | 1.0.0        |
| fastlane-plugin-increment_version_code | 0.4.3        |
</details>
*generated on:* **2021-02-26**
</details>

**Comment by bte74** [comment]
OK, additional info: It looks like the most important part of the error message was somehow not included in the pipeline log. Since the code/IPA builds fine, I tried to upload the ipa using altool. Based on the fact that the initial part of the log is the same, I thought they are using the same engine so running altool from command line could prove helpful in figuring out the issue. Indeed, here's what it tells me:

<<<<<SNIPPED THE BEGINNING OF THE LOG, IT'S THE SAME AS ABOVE>>>>>
[2021-02-26 17:06:52 EET] <main> DBG-X: Apple's web service operation return value:
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter EnableJWTForAllCalls = false
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter MinimumFileSizeThresholdForCheckum = 104857600
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter SoftwareMediaDescriptionFormat = binary
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter RestartClient = false
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter SoftwareMediaDescriptionThresholds = {in-app-purchase-content={extensions=[pkg], size=100000000}, ccats={extensions=[pdf], size=100000000}, product-archive={extensions=[pkg], size=100000000}, developer-id-package={extensions=[zip, pkg, dmg], size=100000000}, bundle={extensions=[zip, ipa], size=100000000}, large-icon={extensions=[jpg, png, jpeg], size=100000000}}
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter Success = false
<<<<<BELOW TEXT WAS NOT AVAILABLE IN THE PIPELINE LOG>>>>>
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter Errors = [ERROR ITMS-90478: "Invalid Version. The build with the version “3.8” can’t be imported because a later version has been closed for new build submissions. Choose a different version number.", ERROR ITMS-90186: "Invalid Pre-Release Train. The train version '3.5' is closed for new build submissions", ERROR ITMS-90062: "This bundle is invalid. The value for key CFBundleShortVersionString [3.5] in the Info.plist file must contain a higher version than that of the previously approved version [3.8]. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring"]
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter FilesToUpload = [metadata.xml, myapp.ipa]
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter AssetsToDescribe = [{Role=source, Filename=myapp.ipa, ShouldDescribeAsset=true, UTI=com.apple.ipa}]
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter Checksums = {myapp.ipa=d27a1beb02b1b162d621149c97edf3fb}
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter FileSizes = {myapp.ipa=159829045}
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter ErrorCode = 1102
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter ErrorMessage = ERROR ITMS-90478: "Invalid Version. The build with the version “3.8” can’t be imported because a later version has been closed for new build submissions. Choose a different version number."
ERROR ITMS-90186: "Invalid Pre-Release Train. The train version '3.5' is closed for new build submissions"
ERROR ITMS-90062: "This bundle is invalid. The value for key CFBundleShortVersionString [3.5] in the Info.plist file must contain a higher version than that of the previously approved version [3.8]. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring
 (1102)
[2021-02-26 17:06:52 EET] <main> DBG-X:   parameter ShouldUseRESTAPIs = false
[2021-02-26 17:06:52 EET] <main> ERROR: ERROR ITMS-90478: "Invalid Version. The build with the version “3.8” can’t be imported because a later version has been closed for new build submissions. Choose a different version number."
[2021-02-26 17:06:52 EET] <main> ERROR: ERROR ITMS-90186: "Invalid Pre-Release Train. The train version '3.5' is closed for new build submissions"
[2021-02-26 17:06:52 EET] <main> ERROR: ERROR ITMS-90062: "This bundle is invalid. The value for key CFBundleShortVersionString [3.5] in the Info.plist file must contain a higher version than that of the previously approved version [3.8]. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring"
[2021-02-26 17:06:52 EET] <main> DBG-X: The error code is: 1102
[2021-02-26 17:06:52 EET] <main>  INFO: Done performing authentication.
[2021-02-26 17:06:52 EET] <main>  INFO: JSON:{"msg":{"phase":"Upload","count":3,"description":"Operation failed","index":3},"messageType":"VerifyProgress"}
[2021-02-26 17:06:52 EET] <main> DBG-X: Returning 1
2021-02-26 17:06:52.539 altool[64474:13169596]  Out:

Now I know what the problem is regarding IPA not being accepted. But the following issues still stand:

1- upload_to_testflight is not picking up the rest of the error log
2- upload_to_testflight is not detecting that the execution of the upload tool is finished and is keeping the pipeline alive, only to be killed out by CICD pipeline timeout
